### PR TITLE
Pushdown string and date formatting functions

### PR DESCRIPTION
**Pull request by iskakaushik**
## Summary

Add query pushdown for five PostgreSQL functions, eliminating local
evaluation for common string manipulation and date formatting patterns:

- **`split_part()`** → `splitByString()` with array indexing (arg reorder)
- **`regexp_replace()`** → `replaceRegexpOne()` / `replaceRegexpAll()` (flag-dependent)
- **`array_to_string()`** → `arrayStringConcat()`
- **`concat_ws()`** → `arrayStringConcat(arrayFilter(...))` with NULL filtering
- **`to_char()`** → `formatDateTime()` with PG→CH format string translation

These functions appear in roughly 25% of surveyed dbt models that
currently fall back to local evaluation.

Closes PG-132, PG-133, PG-134, PG-135.

## Test plan

- [ ] `make tempcheck` passes on local PG 18 + ClickHouse latest
- [ ] CI matrix passes across all supported ClickHouse versions (23.3–25.12)
- [ ] Verify pushdown via `EXPLAIN (VERBOSE, COSTS OFF)` for each function
- [ ] Edge cases: NULL args in `concat_ws`, `MI` vs `MM` ordering in `to_char`, flag variants in `regexp_replace`

### CODE REVIEW

File: test/expected/functions.out [hunk 1]
```diff
@@ -166,10 +166,10 @@ EXPLAIN (VERBOSE, COSTS OFF)
 SELECT coalesce(a::text, b::text, c::text) FROM t1 GROUP BY a, b, c;
```
Comment (serprex) [inline test/expected/functions.out:166]: this should have `ORDER BY 1`

File: test/expected/functions.out [hunk 2]
```diff
@@ -563,7 +563,7 @@ EXPLAIN (VERBOSE, COSTS OFF) SELECT percentile_cont(0.95) WITHIN GROUP (ORDER BY
 SELECT percentile_cont(0.95) WITHIN GROUP (ORDER BY date_part('epoch', timezone('UTC', c))) FROM t1;
  percentile_cont 
 -----------------
-      1546426260
+      1546447860
```
Comment (serprex) [inline test/expected/functions.out]: ?

**Comment by serprex** [comment]
order by should still be there, any query with multiple results which aren't all the same needs `ORDER BY` because SQL doesn't specify order in that circumstance

**Comment by serprex** [comment]
most of these are pushed down now besides to_char, which has gaps:
* `to_char(ts, 'YY')` → PG emits 25, translator emits YY (literal string from CH).
* `to_char(ts, 'Mon DD YYYY')` → PG emits Oct 15 2025, translator emits Mon 15 2025.
* `to_char(ts, 'MM-DD-YY')` → PG emits 10-15-25, translator emits 10-15-YY.
* `to_char(ts, '%Y')` → PG emits %Y, translator emits the year
* double quotes not handled the same between the two
* `to_char(ts, '%Y')` → PG emits HH12 output, translator emits the HH
* incorrect handling of non-const format

correct pushdown needs to check that format is a constant & then validate format is compatible with ClickHouse

closing this PR, a specific to_char implementation can be done separately

**Comment by serprex** [comment]
#244
